src/ui_fsmenu/addons/packager.cc

```diff
@@ -316,26 +316,25 @@ void AddOnsPackager::addon_selected() {
 		return;
 	}
 
-	box_right_subbox_header_hbox_.set_visible(true);
-
 	update_in_progress_ = true;
+
+	box_right_subbox_header_hbox_.set_visible(true);
 	name_.set_text(selected->get_descname());
 	descr_.set_text(selected->get_description());
 	author_.set_text(selected->get_author());
 	version_.set_text(selected->get_version());
 	min_wl_version_.set_text(selected->get_min_wl_version());
 	max_wl_version_.set_text(selected->get_max_wl_version());
-	update_in_progress_ = false;
 
-	if (addon_boxes_.count(selected->get_category()) == 0) {
-		return;
+	if (auto it = addon_boxes_.find(selected->get_category()); it != addon_boxes_.end()) {
```

Comment:
The assignment is not part of the condition, so it should be placed above.
```suggestion
	auto it = addon_boxes_.find(selected->get_category());
	if (it != addon_boxes_.end()) {
```

Reply:
We use init-statements in `if` conditions a lot in other places. They're very useful to limit the scope of a variable that may not hold a meaningful value to the only block where it can be safely used.

https://github.com/widelands/widelands/blob/6eb200ebb7a9f25fbb6c4c6283f3f2e3e772fb7c/src/wlapplication_options.cc#L928
https://github.com/widelands/widelands/blob/6eb200ebb7a9f25fbb6c4c6283f3f2e3e772fb7c/src/logic/addons.cc#L144
https://github.com/widelands/widelands/blob/6eb200ebb7a9f25fbb6c4c6283f3f2e3e772fb7c/src/wui/info_panel.cc#L502